### PR TITLE
Add BLE scanning for S3 and C3.

### PR DESCRIPTION
**Pull request by tannewt**
Everything else should raise NotImplementedError.

First step in #5926

### CODE REVIEW

**Comment by DavePutz** [comment]
Tested on ESP32-S3. Advertisement worked fine, connect did throw a NotImplementedError. So, patch looks good to me.

**Review by dhalbert — COMMENTED** [review]
This looks like it is coming along nicely! TI did not test yet.

I think it would be good to add `// TODO` on every routine that needs implementation, or needs more implementation. That will help us spot what remains to redo.

**Comment by microDev1** [comment]
`esp-idf` was updated recently, a merge from `main` should fix the CI.

**Comment by tannewt** [comment]
> This looks like it is coming along nicely! I did not test yet.
> 
> I think it would be good to add `// TODO` on every routine that needs implementation, or needs more implementation. That will help us spot what remains to redo.

My plan was to search for `NotImplementedError`. Is that enough?

File: ports/espressif/common-hal/_bleio/PacketBuffer.c [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+}
+
+bool common_hal_bleio_packet_buffer_deinited(bleio_packet_buffer_obj_t *self) {
+    return self->characteristic == NULL;
+}
+
+void common_hal_bleio_packet_buffer_deinit(bleio_packet_buffer_obj_t *self) {
+    if (!common_hal_bleio_packet_buffer_deinited(self)) {
+    }
+}
```
Comment (dhalbert) [inline ports/espressif/common-hal/_bleio/PacketBuffer.c:273]: add `// TODO` or raise `NotImplementedError` (if there's anything to do)

**Review by dhalbert — CHANGES_REQUESTED** [review]
I marked some places that should be flagged as NYI.

Also you can add yourself to the copyright on a bunch of files that you're doing a rewrite on,

**Comment by dhalbert** [comment]
> My plan was to search for `NotImplementedError`. Is that enough?

There are some other places too I marked.

**Comment by tannewt** [comment]
Ah, I didn't do a bunch of functions because I just did the construct method. Will add a bunch of TODOs for those spots.

**Comment by tannewt** [comment]
Ok @dhalbert, this is ready for a re-review and passes CI.

**Review by dhalbert — APPROVED** [review]
Great -- thanks!

**Pull request merged**